cheroot/makefile.py

```diff
@@ -30,10 +52,59 @@ def _flush_unlocked(self):
                 # ssl sockets only except 'bytes', not bytearrays
                 # so perhaps we should conditionally wrap this for perf?
                 n = self.raw.write(bytes(self._write_buf))
-            except io.BlockingIOError as e:
-                n = e.characters_written
+            except (io.BlockingIOError, OSError, SysCallError) as e:
+                # Check for a different error attribute depending
+                # on the exception type
+                if isinstance(e, io.BlockingIOError):
+                    n = e.characters_written
```

Comment:
Keep it in a separate `except` block to avoid unnecessary nested blocks. It'll be easier to reason about.